superset/views/dashboard/api.py

```diff
@@ -26,13 +27,32 @@
 
 import superset.models.core as models
 from superset import appbuilder
-from superset.exceptions import SupersetException
+from superset.exceptions import SupersetException, SupersetSecurityException
 from superset.utils import core as utils
 from superset.views.base import BaseSupersetModelRestApi, BaseSupersetSchema
 
+from ..base import check_ownership
 from .mixin import DashboardMixin
 
 
+def check_owner(f):
```

Comment:
Since this is doing a bit more than just "checking owner" (checks for existence of instances), I think you should name it something like `api_security_wrapper`. We could potentially reuse this in other APIs as well. Otherwise, LGTM.

Reply:
Agree, moved it to `views/base.py`